### PR TITLE
Added set_peer_timeout to NetworkedMultiplayerENet. Fixes #20056

### LINKED ISSUE #20056: Expose ENet peer timeout settings
*Bugsquad edit:* This issue is about exposing http://enet.bespin.org/group__peer.html#gac48f35cdd39a89318a7b4fc19920b21b

----

So using the latest godot master branch. I'm working on a multiplayer game. Now sometimes I get about half an hour in to the game and there is a lag spike then I get a disconnect as a result.

It seems as if some packet exceeds its timeout, which causes a disconnect by the server. The client then locks up in PacketPeerUDPPosix::_poll while it waits for recvfrom (which is meant to be non-blocking? so how it can block there is beyond me). After about 30 seconds the client will continue and detect the disconnect. During this time, the game is totally locked up. I can't even get a signal to be emitted that the client is waiting for the server. This should be fixed.

More importantly, there should be a way to change the timeout settings for ENET. It seems a packet has about 1.6 seconds to be acknowledged as being received from the client, so that means its got about 0.8seconds to get tot he client, then 0.8seconds back at most. Any longer and then server will drop them.

Exposing a timeout would allow me to better handle low frame rates, playing over the internet. It would also allow me to easier to debug the network code - I would be able to set a break point and examine code and resume a multiplayer game without the fear of hitting a break point followed by a sudden disconnect.

### PR DESCRIPTION
**Pull request by supagu**
Added set_peer_timeout to NetworkedMultiplayerENet. Fixes #20056

Now for master branch

### CODE REVIEW

File: modules/enet/networked_multiplayer_enet.cpp [hunk 1]
```diff
@@ -807,6 +807,14 @@ int NetworkedMultiplayerENet::get_peer_port(int p_peer_id) const {
 #endif
 }
 
+void NetworkedMultiplayerENet::set_peer_timeout(int p_peer_id, int p_timeout_limit, int p_timeout_min, int p_timeout_max) {
+
+	ERR_FAIL_COND(!peer_map.has(p_peer_id));
+	ERR_FAIL_COND(!is_server() && p_peer_id != 1);
+	ERR_FAIL_COND(peer_map[p_peer_id] == NULL);
+	enet_peer_timeout(peer_map[p_peer_id], p_timeout_limit, p_timeout_min, p_timeout_max);
```
Comment (akien-mga) [inline modules/enet/networked_multiplayer_enet.cpp]: There seems to be trailing whitespace on that line.

**Comment by YeldhamDev** [comment]
A bit of documentation would not hurt.

**Comment by supagu** [comment]
Like code comments or documentation that gets extracted out? If the later, is there an example of how that is formatted some where?

**Comment by akien-mga** [comment]
See https://docs.godotengine.org/en/latest/community/contributing/updating_the_class_reference.html

Basically, `godot --doctool .` using the binary from this PR compiled should update `modules/enet/doc_classes/NetworkedMultiplayerENet.xml` with the method and argument names and an empty `<description>` tag, which you should fill.

**Comment by akien-mga** [comment]
CC @Faless

**Comment by Calinou** [comment]
Remember to add documentation for this method in the class reference :slightly_smiling_face:

**Comment by supagu** [comment]
I haven't had time to lately to add the documentation. If Someone wants to edit this PR to do so go for it.

**Comment by icolwell** [comment]
These timeout values are exactly what I need in my use case, thanks @supagu for creating this PR!
I'd be happy to update the docs, but I'm not sure the best way to go about it? Is it possible to commit directly to this PR? or better/easier to create a new PR based off of this one?

**Comment by akien-mga** [comment]
Superseded by #46470.

**Pull request closed without merging**